File: backend/app/schemas/parent.py
import uuid
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel

from app.schemas.user import UserSummary


class ParentStudentLink(BaseModel):
    parent_id: uuid.UUID
    student_id: uuid.UUID


class ChildSummary(BaseModel):
    student: UserSummary
    course_count: int
    overall_average: Decimal | None
    attendance_rate: Decimal | None


class ParentDigest(BaseModel):
    children: list[ChildSummary]


class ChildProgressDetail(BaseModel):
    student: UserSummary
    courses: list["ChildCourseProgress"]
    upcoming_assignments: list["UpcomingItem"]
    attendance_summary: "ChildAttendanceSummary"


class ChildCourseProgress(BaseModel):
    course_id: uuid.UUID
    course_title: str
    weighted_average: Decimal | None  # 1.0 - 5.0 weighted average
    final_grade: int | None  # rounded to nearest 1-5
    entry_count: int


class UpcomingItem(BaseModel):
    quiz_id: uuid.UUID
    quiz_title: str
    course_title: str
    due_at: datetime | None
    is_submitted: bool


class ChildAttendanceSummary(BaseModel):
    total_days: int
    present: int
    absent: int
    tardy: int
    attendance_rate: Decimal


# Allow forward references
ChildProgressDetail.model_rebuild()

File: frontend/src/types/parent.ts
import type { UserSummary } from './user'

export interface ChildSummary {
  student: UserSummary
  course_count: number
  overall_average: number | null
  attendance_rate: number | null
}

export interface ParentDigest {
  children: ChildSummary[]
}

export interface ChildCourseProgress {
  course_id: string
  course_title: string
  weighted_average: number | null
  final_grade: number | null
  entry_count: number
}

export interface UpcomingItem {
  quiz_id: string
  quiz_title: string
  course_title: string
  due_at: string | null
  is_submitted: boolean
}

export interface ChildAttendanceSummary {
  total_days: number
  present: number
  absent: number
  tardy: number
  attendance_rate: number
}

export interface ChildProgressDetail {
  student: UserSummary
  courses: ChildCourseProgress[]
  upcoming_assignments: UpcomingItem[]
  attendance_summary: ChildAttendanceSummary
}